src/main/java/com/google/api/codegen/transformer/RetryDefinitionsTransformer.java

```diff
@@ -77,23 +101,141 @@
                   interfaceConfigProto.getName()));
         }
       }
-      builder.put(retryDef.getName(), (new ImmutableSet.Builder<String>()).addAll(codes).build());
+      builder.put(retryDef.getName(), (new ImmutableList.Builder<String>()).addAll(codes).build());
+    }
+    if (diagCollector.getErrorCount() > 0) {
+      return null;
+    }
+    return builder.build();
+  }
+
+  /**
+   * Returns a mapping of a retryCodeDef name to the list of retry codes it contains. Also populates
+   * the @param methodNameToRetryCodeNames with a mapping of a Method name to its retry code
+   * settings nmame.
+   */
+  private static ImmutableMap<String, List<String>> createRetryCodesDefinition(
+      DiagCollector diagCollector,
+      Interface apiInterface,
+      ImmutableMap.Builder<String, String> methodNameToRetryCodeNames,
+      ProtoParser protoParser,
+      SymbolTable symbolTable) {
+
+    ImmutableMap.Builder<String, List<String>> builder = ImmutableMap.builder();
+
+    // Check proto annotations for retry settings.
+    for (Method method : apiInterface.getMethods()) {
+
+      Retry retry = protoParser.getRetry(method);
+
+      Set<String> retryCodes = new TreeSet<>();
+      if (protoParser.isHttpGetMethod(method)) {
+        // If this is analogous to HTTP GET, then automatically retry on `INTERNAL` and
+        // `UNAVAILABLE`.
+        retryCodes.addAll(RETRY_CODES_FOR_HTTP_GET);
```

Comment:
The "Implementing API Code Generator" doc says "If the google.api.retry annotation is present on the RPC, then the error codes specified in the codes key should be retried". I think this implies we should replace the retry code set, not add to it?